### ISSUE TITLE
[Feature]: Adding Engaging Visual Elements/Map in Documentation

### ISSUE THREAD

**Issue by Fe-r-oz** [aarontrowbridge/QuantumCollocation.jl#94]
### Feature Description

![Collocation5](https://github.com/aarontrowbridge/QuantumCollocation.jl/assets/93876775/b0f82a86-1d1e-44a1-b180-35db08f7b7ca)

### Importance

4

### What does this feature affect?

- [ ] quantum system construction
- [ ] problem setup
- [ ] problem solution
- [ ] problem performance
- [ ] solution analysis
- [ ] plotting
- [X] documentation
- [ ] tests
- [ ] other (please specify below)

### Other information

Adding schematics like this allows the beginners to get a complete overview of the package in an instant. It also makes the documentation more engaging to read without having to explore the API.

**Comment by aarontrowbridge** [comment]
hey @Fe-r-oz sorry we are just seeing this, we had something like this before, but this is MUCH clearer, we will probably merge it in later this week

**Comment by Fe-r-oz** [comment]
Hi @aarontrowbridge , No problem.  I would like to have your thoughts on another approach. Let's call it a docstring based approach instead of a hard coded figure based approach.

I think in the long run, a better, maintainable solution would be by using dynamic schematics created in docstring using Mermaid. This means that whenever a new functionality is added, we can just add update the code with ease instead of using hard coded schematics. 

Here is a rough version with default color scheme in Mermaid attached below. As it's a screenshot attached below, it's not clear, But it will be much clearer when rendered via browser.  

This approach might not be as fancy as the one above but it's maintainable and easily updateable. What do you suggest? 

I can submit a polished PR of this approach later this week. Please do let me know if there is any other flowcharts we can add, refer me to the paper (if any) and I will look into it as well

![Screenshot 2024-05-29 at 19-43-48 Screenshot](https://github.com/aarontrowbridge/QuantumCollocation.jl/assets/93876775/7213bf22-71c1-4b02-a7ab-8ed8c8879fc9)

**Comment by andgoldschmidt** [comment]
Mermaid seems to flatten the hierarchical structure of the package. @Fe-r-oz Is this expected, or an artifact of the screenshot? For example, we see here that `ProblemTemplate` and `QuantumUtils` are repeated many times, but in the original post the hierarchy looks correct and easy to interpret.

**Comment by Fe-r-oz** [comment]
Indeed, we don't want the labels to be repeated many times. Thanks for pointing that out. I have fixed it.

How about this (screenshot taken from browser)?

![Screenshot from 2024-05-29 23-49-44](https://github.com/aarontrowbridge/QuantumCollocation.jl/assets/93876775/c20b6086-387d-4fbc-b50b-517fbfe3e74a)

**Comment by andgoldschmidt** [comment]
That looks good! Is there a way to tag what functions appear in the preview? Perhaps, a flag in the docstring parsed by the dynamic schematic that says "yes, this function should be displayed". 

My reason is that not all exported functions have the same priority for users. For example, `isovec_unitary_fidelity` is useful to export, but most users will interact with `unitary_fidelity`.

**Comment by Fe-r-oz** [comment]
Option 1
![Screenshot from 2024-05-30 12-33-09](https://github.com/aarontrowbridge/QuantumCollocation.jl/assets/93876775/a24984c5-0160-42e2-9f40-eec02c8d36ed)

Option 2

![Screenshot from 2024-05-30 12-33-22](https://github.com/aarontrowbridge/QuantumCollocation.jl/assets/93876775/aad112f0-9a54-4f2f-91e3-95044c07aa14)

Option 3: We can only include `unitary fidelity`. 

Mermaid flowcharts is typically written in HTLML format  `<div class ="Mermaid"> ...</div> `so there is no explicit tag. Nevertheless, I have presented 3 options which are close to your requirements. Please let me know which sounds good.

**Comment by aarontrowbridge** [comment]
I think `isovec_unitary_fidelity` should be treated as a unique function. This looks good though!

**Comment by Fe-r-oz** [comment]
Thanks for your suggestion.

In the ` """ [docs](url)  """,`  I think we can add the href with `[Preferred Title](url)` similar to how we do it in .md.
